aptos-move/framework/aptos-framework/sources/SharedAccount.move

```diff
@@ -0,0 +1,153 @@
+module AptosFramework::SharedAccount {
+    use Std::Errors;
+    use Std::Signer;
+    use Std::Vector;
+    use AptosFramework::Account;
+    use AptosFramework::Coin;
+    use AptosFramework::SimpleMap::{Self, SimpleMap};
+    use AptosFramework::ResourceAccount;
+
+    struct Shares has store {
+        numerator: u64,
+        denominator: u64,
+    }
+
+    // Resource representing a shared account
+    struct SharedAccount has key {
+        share_record: SimpleMap<address, Shares>,
+    }
+
+    const EINVALID_INIT_INPUT: u64 = 0;
+    const EACCOUNT_DNE: u64 = 1;
+    const EADDRESS_ALREADY_EXISTS: u64 = 2;
+    const EINVALID_NUMERATOR_DENOMINATOR_COMBINATIONS: u64 = 3;
+    const EINVALID_SIGNER: u64 = 4;
+    const EINSUFFICIENT_BALANCE: u64 = 5;
```

Comment:
At one point we pulled error messages out of the error code comments, but I don't think we do that anymore?  @movekevin